### DISCUSSION TITLE
Image magick 6.8 creating multiple renditions for certain pdfs

### DISCUSSION THREAD

**Discussion by veenavnair** [ImageMagick/ImageMagick#3285] [Development]
We are trying to create a web jpg rendition of the pdf using below Image magick command.

convert "test.pdf" -resize 480x384> -colorspace rgb test.jpg

That time it is creating jpg renditions for each page of the pdf where as for other pdfs only for the 1st page it is creating a jpg rendition.I would like to know what is the default behaviour for Image magick and why it is behaving differently for each of those pdfs.Please advise.

**Comment by snibgo** [comment]
In your command:

"test.pdf" will create one image per PDF page.

-resize 480x384> will resize all the images

-colorspace rgb will convert all the images to linear RGB. I suspect you would prefer sRGB.

test.jpg will write the images. If there is only one, it will be written to "test.jpg". If there is more than one, it will be written to "test-0.jpg", test-1.jpg", etc.

**Reply by veenavnair** [reply]
[test2.pdf](https://github.com/ImageMagick/ImageMagick/files/6024486/test2.pdf)
![test2_imconvert](https://user-images.githubusercontent.com/78795421/108755951-ea6f9b80-756d-11eb-81bf-9653b2fdf972.jpg)
![test1_imconvert](https://user-images.githubusercontent.com/78795421/108755959-ecd1f580-756d-11eb-924f-408b494cad38.jpg)
![sample_imconvert](https://user-images.githubusercontent.com/78795421/108755960-ed6a8c00-756d-11eb-94fe-03d5137d8a2e.jpg)
[test1.pdf](https://github.com/ImageMagick/ImageMagick/files/6024487/test1.pdf)
[sample.pdf](https://github.com/ImageMagick/ImageMagick/files/6024488/sample.pdf)






@snibgo Thanks for your reply.Attached are few files I tried which is giving only one jpg rendition for 1st page.For rest of the pages it is not creating renditions.eg: test1.pdf is the input and test1_imconvert.jpg is the output generated with below command.
convert "test1.pdf" -resize 480x384> -colorspace rgb test1_imconvert.jpg
convert "test1.pdf" -resize 480x384> -colorspace sRGB test1_imconvert.jpg
Can you help me find out why it is not creating renditions for each page as it is supposed to be?

**Comment by snibgo** [comment]
What version of IM? On what platform? You may need to escape the ">" character.

With IM v7.0.8-64 on Windows 8.1 this works fine:
```
magick test1.pdf -resize "480x384^>" -colorspace rgb abc.jpg
```
It makes files abc-0.jpg to abc-57.jpg.

If you use IM v7, I suggest you use "magick", not "convert".

What is the text output from these two commands:
```
magick -verbose test1.pdf[0] info:

magick test1.pdf info:
```

**Reply by veenavnair** [reply]
We are using IM 6.8.0 version,windows 2012 server."magick: command is not existing.so tried below ones.Results displayed in the screenshot are in respective order.Executable file name is "imconvert".
1. imconvert -verbose "C:\Users\vnai0488\Pictures\test\sample.pdf"[0] info:
2. imconvert "C:\Users\vnai0488\Pictures\test\sample.pdf" info:
![image](https://user-images.githubusercontent.com/78795421/108839168-b55d5a80-75fa-11eb-8ede-3b815d8f42fb.png)

Picture convert command used: imconvert "C:\Users\vnai0488\Pictures\test\sample.pdf" -resize 480x384> -colorspace rgb C:\Users\vnai0488\Pictures\test\test1_imconvert.jpg

**Comment by snibgo** [comment]
Your command "imconvert sample.pdf info:" shows only one image. That is why you get only one jpeg. How do you know that sample.pdf has more than one page?

Your IM is old, 6.8.0-5. Is your Ghostscript also old? Upgrading those products might help.

**Reply by veenavnair** [reply]
Sample.pdf has 5 pages,I opened it and checked.It is the same one which I attached previously to the ticket.Attaching here again
[sample.pdf](https://github.com/ImageMagick/ImageMagick/files/6042843/sample.pdf).Ghostscript 9.15 is used.

**Reply by veenavnair** [reply]
Upgrading IM cannot be done for now as we are using it with another application (OTMM,It supports only 6.x version.Any workaround to make it work with current setup?

**Comment by snibgo** [comment]
According to IM v6.9.9-50 and v7.0.8-64, with Ghostscript v9.19, your PDF (renamed "sample4.pdf" on my computer) has 5 pages. 
```
f:\web\im>%IM%convert sample4.pdf info:
sample4.pdf[0] PDF 612x792 612x792+0+0 16-bit sRGB 63247B 0.219u 0:00.265
sample4.pdf[1] PDF 612x792 612x792+0+0 16-bit sRGB 63247B 0.188u 0:00.265
sample4.pdf[2] PDF 612x792 612x792+0+0 16-bit sRGB 63247B 0.141u 0:00.235
sample4.pdf[3] PDF 612x792 612x792+0+0 16-bit sRGB 63247B 0.125u 0:00.235
sample4.pdf[4] PDF 612x792 612x792+0+0 16-bit sRGB 63247B 0.094u 0:00.218

f:\web\im>%IMG7%magick sample4.pdf info:
sample4.pdf[0] PDF 612x792 612x792+0+0 16-bit sRGB 63247B 0.813u 0:00.277
sample4.pdf[1] PDF 612x792 612x792+0+0 16-bit sRGB 97692B 0.781u 0:00.276
sample4.pdf[2] PDF 612x792 612x792+0+0 16-bit sRGB 100348B 0.594u 0:00.239
sample4.pdf[3] PDF 612x792 612x792+0+0 16-bit sRGB 73332B 0.453u 0:00.199
sample4.pdf[4] PDF 612x792 612x792+0+0 16-bit sRGB 72047B 0.313u 0:00.163
```
You might try upgrading Ghostscript. I don't know what else to suggest.

**Reply by veenavnair** [reply]
I will try with updated ghostscript.Thanks for your response :-) !
